projects/storefrontlib/shared/services/file/files-form-validators.ts

```diff
@@ -56,4 +56,30 @@ export class FilesFormValidators {
       return Object.keys(errors).length === 0 ? null : errors;
     };
   }
+
+  /**
+   * Checks allowed types
+   *
+   * @param {Array<string>} allowedTypes Allowed types of files
+   * @returns Uses 'notParsable' validator error with allowedTypes property
+   * @memberOf FilesFormValidators
+   */
+  allowedTypes(allowedTypes?: Array<string>): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors: ValidationErrors = {};
+      if (allowedTypes && control.value) {
+        const files: File[] = Array.from(control.value);
+        files.forEach((file: File) => {
+          if (!allowedTypes.includes(this.getExtension(file.name))) {
+            errors.notParsable = true;
+          }
+        });
+      }
+      return Object.keys(errors).length === 0 ? null : errors;
+    };
+  }
+
+  protected getExtension(filename?: string): string {
+    return `.${filename?.split('.').pop()}` || '';
+  }
```

Comment:
Why filename is optional?
The return will never fallback in. ->.  || '';
Since the worst case scenario `.${filename?.split('.').pop()}` will be '.undefined'

In addition, if the filename does not have a '.' character the result of `.${filename?.split('.').pop()}` will be the same filename provided that the filename is not undefined
```  this.getExtension('filewithoutextension');   ==>  '.filewithoutextension' ```

Reply:
let's discuss in a call